tests/test_optimizer/test_dist_adafactor.py

```diff
@@ -0,0 +1,626 @@
+
+    # base_param_group = setup_param_groups([weight, bias])
+    # cp_param_group = setup_param_groups([weight_col_shard_flatten, bias_col_flatten])
+    # rp_param_group = setup_param_groups([weight_row_shard_flatten, bias_row_flatten])
+
+    # ==============================
+    # Init Optimizer
+    # ==============================
+
+    # base
+    optimizer_base = Adafactor([weight, bias])
+    cp_dist_optim = DistributedAdaFactor([weight_col_shard_flatten, bias_col_flatten])
+    rp_dist_optim = DistributedAdaFactor([weight_row_shard_flatten, bias_row_flatten])
+
+    shard_to_param_cp = set_master_param_to_shard_param([weight_col_shard_flatten, bias_col_flatten])
+    cp_dist_optim.setup_distributed(
+        tensor_parallel_group=tp_group,
+        data_parallel_group=dp_group,
+        shard_to_param=shard_to_param_cp,
+        use_zero=use_zero,
+    )
+
+    shard_to_param_rp = set_master_param_to_shard_param([weight_row_shard_flatten, bias_row_flatten])
+    rp_dist_optim.setup_distributed(
+        tensor_parallel_group=tp_group,
+        data_parallel_group=dp_group,
+        shard_to_param=shard_to_param_rp,
+        use_zero=use_zero,
+    )
+
+    N_STEPS = 1
+    for _ in range(N_STEPS):
+        # base step
+        optimizer_base.zero_grad()
+        weight.grad = torch.rand_like(weight)
+        bias.grad = torch.rand_like(bias)
+        optimizer_base.step()
+
+        # col parallel step
+        cp_dist_optim.zero_grad()
+        weight_col_shard_flatten.grad = (
+            distribute_tensor(weight.grad, get_device_mesh(weight_col_shard), weight_col_shard_shard_spec)
+            .clone()
+            .flatten()
+        )
+        bias_col_flatten.grad = bias.grad.clone().flatten()
+        cp_dist_optim.step()
+
+        # row parallel step
+        rp_dist_optim.zero_grad()
+        weight_row_shard_flatten.grad = (
+            distribute_tensor(weight.grad, get_device_mesh(weight_row_shard), weight_row_shard_shard_spec)
+            .clone()
+            .flatten()
+        )
+        bias_row_flatten.grad = bias.grad.clone().flatten()
+        rp_dist_optim.step()
+
+        # gather result
+        weight_col_gather = _gather(
+            input_=weight_col_shard_flatten.data.view(-1, H // tp_size),
+            dim=-1,
+            process_group=tp_group,
+        )  # gather
+        weight_row_gather = _gather(input_=weight_row_shard_flatten.data, dim=-1, process_group=tp_group).view(
+            -1, W
+        )  # gather
+
+        # verify
+        correctness_verify(weight.data, weight_col_gather.data, dtype)
+        correctness_verify(weight.data, weight_row_gather.data, dtype)
+
+    print(f"Base Test Pass")
+
+
+@parameterize("dtype", [torch.float16])  # torch.float32, torch.float16, torch.bfloat16
+@parameterize("tp_zero_size", [(1, 4)])  # (2, 2), (4, 1), (1, 4)
+def exam_dist_adafactor_zero(dtype: torch.dtype, tp_zero_size: tuple[int, int]):
```

Comment:
As some parts of this functions are tested in `exam_dist_adafactor_base()` and `exam_bert_test()`, can we remove this?

Reply:
> As some parts of this functions are tested in `exam_dist_adafactor_base()` and `exam_bert_test()`, can we remove this?

The zero test does not appear in base, so I consider it to be the next level of test in base, containing zero (but not tp); 
And the bert test contains tests for HybridPlugin (tp+zero) and LowLevelZeroPlugin;